src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,284 @@
+      this.targetAngle = targetAngle;
+      this.angleTolerance = angleTolerance;
+    }
+
+    ArmPosition(Angle targetAngle) {
+      this(targetAngle, Degrees.of(2)); // 2 degree default tolerance
+    }
+  }
+
+  /**
+   * Gets the current arm position mode.
+   *
+   * @return The current ArmPosition
+   */
+  public ArmPosition getMode() {
+    return currentMode;
+  }
+
+  /**
+   * Sets a new arm position and schedules the corresponding command.
+   *
+   * @param position The desired ArmPosition
+   */
+  public void setArmPosition(ArmPosition position) {
+    currentCommand.cancel();
+    currentMode = position;
+    currentCommand.schedule();
+  }
+
+  // Command that runs the appropriate routine based on the current position
+  private final Command currentCommand =
+      new SelectCommand<>(
+          Map.of(
+              ArmPosition.STOP,
+              Commands.runOnce(this::stop)
+                  .alongWith(Commands.run(() -> checkAtTarget(ArmPosition.STOP)))
+                  .withName("Stop Arm"),
+              ArmPosition.STOWED,
+              createPositionCommand(ArmPosition.STOWED),
+              ArmPosition.INTAKE,
+              createPositionCommand(ArmPosition.INTAKE),
+              ArmPosition.AMP,
+              createPositionCommand(ArmPosition.AMP),
+              ArmPosition.SPEAKER,
+              createPositionCommand(ArmPosition.SPEAKER)),
+          this::getMode);
+
+  /**
+   * Creates a command for a specific arm position that moves the arm and checks the target
+   * position.
+   *
+   * @param position The arm position to create a command for
+   * @return A command that implements the arm movement
+   */
+  private Command createPositionCommand(ArmPosition position) {
+    return Commands.parallel(
+        Commands.runOnce(() -> setPosition(position.targetAngle))
+            .withName("Move to " + position.toString()),
+        Commands.run(() -> checkAtTarget(position))
+            .withName("Check " + position.toString() + " Target"));
+  }
+
+  /**
+   * Checks if the arm is at its target position.
+   *
+   * @return true if at target position, false otherwise
+   */
+  @AutoLogOutput
+  private boolean isAtTarget() {
+    if (currentMode == ArmPosition.STOP) return true;
+    return getPosition().isNear(currentMode.targetAngle, currentMode.angleTolerance);
+  }
+
+  /**
+   * Logs whether the arm is at its target position for a given mode.
+   *
+   * @param position The position to check against
+   */
+  private void checkAtTarget(ArmPosition position) {
+    boolean atTarget = isAtTarget();
+    Logger.recordOutput("Arm/AtTarget", atTarget);
+    Logger.recordOutput("Arm/TargetAngle", position.targetAngle);
+  }
+
+  /**
+   * Creates a command to set the arm to a specific position.
+   *
+   * @param position The desired arm position
+   * @return Command to set the position
+   */
+  public Command setPositionCommand(ArmPosition position) {
```

Comment:
makei this private and force consumers to use your factory methods. 
i think this is especially important since you haven't incorporated any limit switches.